app/src/org/commcare/android/framework/SessionAwareFragmentActivity.java

```diff
@@ -0,0 +1,25 @@
+package org.commcare.android.framework;
+
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+
+/**
+ * Manage redirection to login screen when session expiration occurs.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public abstract class SessionAwareFragmentActivity extends FragmentActivity {
```

Comment:
Is there some fancy thing I can do to avoid duplicating the code here in SessionAwarePreferenceActivity as well?